README.rst
Revision Control - All-in-one code repository
=============================================

An integrated revision control server combining the world's best open
source Version Control Systems: Subversion and Git.
A web interface for each system is included, making it easy to browse
through the code base, compare revisions and manage repositories for
multiple projects.

This appliance includes all the standard features in `TurnKey Core`_,
and on top of that:

- Revision Control systems supported: Git, Subversion.
   
   - Includes exemplary helloworld repositories.
   - Includes GitHub inspired theme by kogakure for gitweb.

- SSL support out of the box.
- Webmin module for configuring Apache2.
- Includes TurnKey web control panel (convenience).

- Repository access::

    Name        Web frontend    Web access          Protocol access
    ----        ------------    ----------          ---------------
    Git         gitweb          http://addr/git     vcs@addr:/git, http://addr/git, https://addr/git
    Subversion  viewvc          http://addr/svn     svn://addr/svn, svn+ssh://addr/srv/repos/svn
    Repositories are stored in /srv/repos.

Credentials *(passwords set at first boot)*
-------------------------------------------

-  Webmin, Webshell, SSH: username **root**
-  Git, Subversion: username **vcs**


.. _TurnKey Core: https://www.turnkeylinux.org/core


changelog
turnkey-revision-control-17.1 (1) turnkey; urgency=low

  * Updating revision-control and its dependecies to v17.

  * Downloading the ViewVC package from the upstream github repository.

  * Note: Please refer to turnkey-core's 17.1 changelog for changes common to all
    appliances. Here we only describe changes specific to this appliance.

 -- Mattie Darden <mattie@turnkeylinux.org>  Sat, 21 Jan 2023 18:37:47 +0000

turnkey-revision-control-16.1 (1) turnkey; urgency=low

  * Rebuild on latest Debian Buster.

  * Note: Please refer to turnkey-core's 16.1 changelog for changes common to all
    appliances. Here we only describe changes specific to this appliance.

 -- Jeremy Davis <jeremy@turnkeylinux.org>  Tue, 11 May 2021 12:07:09 +1000

turnkey-revision-control-16.0 (1) turnkey; urgency=low

  * Explcitly disable TLS<1.2 (i.e. SSLv3, TLSv1, TLSv1.1). (v15.x
    TurnKey releases supported TLS 1.2, but could fallback as low as TLSv1).

  * Update SSL/TLS cyphers to provide "Intermediate" browser/client support
    (suitable for "General-purpose servers with a variety of clients,
    recommended for almost all systems"). As provided by Mozilla via
    https://ssl-config.mozilla.org/.

  * Note: Please refer to turnkey-core's changelog for changes common to all
    appliances. Here we only describe changes specific to this appliance.

 -- Zhenya Hvorostian <zhenya@turnkeylinux.org>  Fri, 14 Aug 2020 19:04:10 +0300

turnkey-revision-control-15.0 (1) turnkey; urgency=low

  * Remove Bazaar (bzr) & Mercurial (hg) (part of #1050).
  * Replace WebSVN with ViewVC (part of #1050).
  * Install git-web theme properly (closes #979).
  * Update git config and documentation (closes #980).
  * Change default user and /srv/repos owner to 'vcs'.
  * Provide tkl-webcp in Perl (so PHP no longer required).
    [ Anton Pyrogovskyi ]

  * Note: Please refer to turnkey-core's changelog for changes common to all
    appliances. Here we only describe changes specific to this appliance.

 -- Jeremy Davis <jeremy@turnkeylinux.org>  Fri, 15 Jun 2018 14:25:14 +1000

turnkey-revision-control-14.2 (1) turnkey; urgency=low

  * Fixed Smart HTTP misconfiguration [#407].

  * Note: Please refer to turnkey-core's changelog for changes common to all
    appliances. Here we only describe changes specific to this appliance.

 -- Anton Pyrogovskyi <anton@turnkeylinux.org>  Sat, 27 May 2017 06:21:51 +0200

turnkey-revision-control-14.1 (1) turnkey; urgency=low

  * Installed security updates.

  * Installed updated packages from TurnKey repo
      - includes relevant Webmin packages (v1.780)

  * Note: Please refer to turnkey-core's changelog for changes common to all
    appliances. Here we only describe changes specific to this appliance.

 -- Jeremy Davis <jeremy@turnkeylinux.org>  Tue, 01 Mar 2016 10:23:54 +0000

turnkey-revision-control-14.0 (1) turnkey; urgency=low

  * Latest Debian Wheezy package versions of all components.
   
    - removed bzr-rebase (Debian package no longer available).
    - multiple minor bug fixes.

   * Hardened default SSL settings
   
   * Note: Please refer to turnkey-core's changelog for changes common to all
     appliances. Here we only describe changes specific to this appliance.

 -- Jeremy Davis <jeremy@turnkeylinux.org>  Fri, 05 Jun 2015 23:54:45 +1000

turnkey-revision-control-13.0 (1) turnkey; urgency=low

  * Git:

    - Updated gitweb-force-texthtml patch [#81].
    - Gitweb static files are now in static/ [#104].

  * Bazaar:

    - Specify loggerhead bzr_root as absolute url [#116].

  * Note: Please refer to turnkey-core's changelog for changes common to all
    appliances. Here we only describe changes specific to this appliance.

 -- Alon Swartz <alon@turnkeylinux.org>  Fri, 11 Oct 2013 11:04:45 +0300

turnkey-revision-control-12.1 (1) turnkey; urgency=low

  * Latest version of github inspired theme by kogakure for gitweb will
    be installed at build time.

  * Note: Please refer to turnkey-core's changelog for changes common to all
    appliances. Here we only describe changes specific to this appliance.

 -- Alon Swartz <alon@turnkeylinux.org>  Sun, 07 Apr 2013 08:00:00 +0200

turnkey-revision-control-12.0 (1) turnkey; urgency=low

  * Added github inspired theme by kogakure to gitweb.
  * Replaced git-daemon-run with git-daemon-sysvinit.
  * Removed typo from mercurial helloworld hgrc (bugfix).
  * Fixed example mercurial clone address: hg -> http (bugfix).

  * Major component versions

    git-core                1:1.7.2.5-3
    gitweb                  1:1.7.2.5-3
    kogakure-gitweb-theme   revision 4f3a048 (upstream archive)
    bzr                     2.1.2-1
    loggerhead              1.17+bzr424-1
    subversion              1.6.12dfsg-6
    websvn                  2.3.1-1
    mercurial               1.6.4-1
    build-essential         11.5
    apache2                 2.2.16-6+squeeze7

  * Note: Please refer to turnkey-core's changelog for changes common to all
    appliances. Here we only describe changes specific to this appliance.

 -- Alon Swartz <alon@turnkeylinux.org>  Wed, 01 Aug 2012 08:00:00 +0200

turnkey-revision-control-11.3 (1) turnkey; urgency=low

  * Installed security updates.
  * Enabled etckeeper garbage collection by default.
  * Upgraded to latest inithooks version (adhoc re-initialization via turnkey-init)

 -- Alon Swartz <alon@turnkeylinux.org>  Mon, 05 Dec 2011 10:48:44 +0000

turnkey-revision-control-11.2 (1) turnkey; urgency=low

  * Installed security updates.
  * Added HubDNS package and firstboot configuration.

 -- Alon Swartz <alon@turnkeylinux.org>  Fri, 15 Jul 2011 07:47:08 +0000

turnkey-revision-control-11.1 (1) turnkey; urgency=low

  * Transitioned to upstream version of loggerhead.
  * Still using custom loggerhead init script (flexibility, stability).

  * Major component versions:

    git-core        1:1.7.0.4-1ubuntu0.1
    gitweb          1:1.7.0.4-1ubuntu0.1
    bzr             2.1.1-1

overlay/var/www/index.php
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">

<html lang="en">
    <head>
        <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
        <meta http-equiv="Content-Style-Type" content="text/css">
        <meta http-equiv="Content-Script-Type" content="text/javascript">

        <title>TurnKey Revision Control</title>
        
        <link rel="stylesheet" href="css/ui.tabs.css" type="text/css" media="print, projection, screen">
        <link rel="stylesheet" href="css/base.css" type="text/css">

        <script src="js/jquery-1.2.6.js" type="text/javascript"></script>
        <script src="js/ui.core.js" type="text/javascript"></script>
        <script src="js/ui.tabs.js" type="text/javascript"></script>
        <script type="text/javascript">
            $(function() {
                $('#container-1 > ul').tabs({ fx: { opacity: 'toggle'} });
            });
        </script>
    </head>

    <body>
        <h1>TurnKey Revision Control</h1>
        
        <div id="container-1">
            <ul>
                <li><a href="#cp"><span>Control Panel</span></a></li>
                <li><a href="#examples"><span>Examples</span></a></li>
            </ul>

            <div id="cp">
                <div class="fragment-content">
                    <div>
                        <a href="/git"><img
                        src="images/git.png"/>Git</a>
                    </div>
                    <div>
                        <a href="/svn"><img
                        src="images/subversion.png"/>Subversion</a>
                    </div>
                    <div>
                        <a href="https://<?php print
                        $_SERVER{'HTTP_HOST'}; ?>:12320"><img
                        src="images/shell.png"/>Web Shell</a>
                    </div>
                    <div>
                        <a href="https://<?php print
                        $_SERVER{'HTTP_HOST'}; ?>:12321"><img
                        src="images/webmin.png"/>Webmin</a>
                    </div>
                    <br />

                    <h2>Resources and references</h2>
                    <ul>
                        <li>
                          <b>Official project websites:</b>
                          <a href="http://git-scm.com">Git</a>, 
                          <a href="http://subversion.tigris.org">Subversion</a>
                        </li>
                        <li>
                          <b>Local documentation:</b>
                          <a href="/docs/git">Git</a>,
                          <a href="/docs/svn">Subversion</a>
                        </li>
                        <li>
                          <b>Wikipedia on revision control:</b> <a href="http://en.wikipedia.org/wiki/Revision_control">Overview</a>, <a
href="http://en.wikipedia.org/wiki/Distributed_revision_control">Centralized
vs. Distributed</a>, <a href="http://en.wikipedia.org/wiki/Comparison_of_revision_control_software">Comparisons</a> 
                        </li>
                        <li><a href="https://www.turnkeylinux.org/revision-control">TurnKey Revision Control release notes</a></li>
                    </ul>

                </div>
            </div>

            <div id="examples">
                <div class="fragment-content">

<h1>Clone/checkout repository (helloworld)</h1>

<h2>Git</h2>
<pre>git clone git://<?php print $_SERVER{'HTTP_HOST'}; ?>/git/helloworld</pre>

<h2>Subversion</h2>
<pre>svn co svn://<?php print $_SERVER{'HTTP_HOST'}; ?>/svn/helloworld</pre>

                </div>
            </div>

        </div>
    </body>
</html>


overlay/var/www/index.pl
#!/usr/bin/perl

print <<EOF;
Content-Type: text/html; charset=utf-8


<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">

<html lang="en">
    <head>
        <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
        <meta http-equiv="Content-Style-Type" content="text/css">
        <meta http-equiv="Content-Script-Type" content="text/javascript">

        <title>TurnKey Revision Control</title>
        
        <link rel="stylesheet" href="css/ui.tabs.css" type="text/css" media="print, projection, screen">
        <link rel="stylesheet" href="css/base.css" type="text/css">

        <script src="js/jquery-1.2.6.js" type="text/javascript"></script>
        <script src="js/ui.core.js" type="text/javascript"></script>
        <script src="js/ui.tabs.js" type="text/javascript"></script>
        <script type="text/javascript">
            \$(function() {
                \$('#container-1 > ul').tabs({ fx: { opacity: 'toggle'} });
            });
        </script>
    </head>

    <body>
        <h1>TurnKey Revision Control</h1>
        
        <div id="container-1">
            <ul>
                <li><a href="#cp"><span>Control Panel</span></a></li>
                <li><a href="#examples"><span>Examples</span></a></li>
            </ul>

            <div id="cp">
                <div class="fragment-content">
                    <div></div>
                    <div>
                        <a href="/git"><img
                        src="images/git.png"/>Git</a><br />
                        <font size="-4">
                        via <a href="https://git-scm.com/book/en/v2/Git-on-the-Server-GitWeb">GitWeb</a>
                        </font>
                    </div>
                    <div>
                        <a href="/svn"><img
                        src="images/subversion.png"/>Subversion</a><br />
                        <font size="-4">
                        via <a href="http://viewvc.org/">ViewVC</a>
                        </font>
                    </div>
                    <div>
                        <a href="https://$ENV{HTTP_HOST}:12320"><img
                        src="images/shell.png"/>Web Shell</a>
                    </div>
                    <div>
                        <a href="https://$ENV{HTTP_HOST}:12321"><img
                        src="images/webmin.png"/>Webmin</a>
                    </div>
                    <div></div>
                    <h2>Resources and references</h2>
                    <ul>
                        <li>
                          <b>Official project websites:</b>
                          <a href="https://git-scm.com">Git</a>,
                          <a href="https://subversion.apache.org/">Subversion</a>
                        </li>
                        <li>
                          <b>Local documentation:</b>
                          <a href="/docs/git">Git</a>,
                          <a href="/docs/svn">Subversion</a>
                        </li>
                        <li>
                          <b>Wikipedia on revision control:</b> <a href="http://en.wikipedia.org/wiki/Revision_control">Overview</a>, <a
href="http://en.wikipedia.org/wiki/Distributed_revision_control">Centralized
vs. Distributed</a>, <a href="http://en.wikipedia.org/wiki/Comparison_of_revision_control_software">Comparisons</a> 
                        </li>
                        <li><a href="https://www.turnkeylinux.org/revision-control">TurnKey Revision Control release notes</a></li>
                    </ul>

                </div>
            </div>

            <div id="examples">
                <div class="fragment-content">
                    <h1>Clone/checkout repository (helloworld)</h1>

                    <h2>Git via SSH</h2>
                    <pre>git clone ssh://vcs\@$ENV{'HTTP_HOST'}/~/git/helloworld</pre>

                    <h2>Subversion via SVN</h2>
                    <pre>svn co svn://$ENV{'HTTP_HOST'}/svn/helloworld</pre>
                </div>
            </div>

        </div>
    </body>
</html>
EOF
